subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationDescriptor.java

```diff
@@ -21,21 +21,27 @@
 /**
  * Meta-data about a build operation.
  */
-public class BuildOperationDetails {
-    private final BuildOperationExecutor.Operation parent;
+public class BuildOperationDescriptor {
+    private final Long id;
```

Comment:
I was just reading this comment: https://github.com/gradle/gradle/pull/1838#pullrequestreview-33076390
...which suggests otherwise.

There is at least one place outside of BuildOperationExecutor that creates OperationIdentifiers.
(https://github.com/gradle/gradle/blob/dc6eb2da25ffbe73e6fe5497413946e2c2512058/subprojects/logging/src/main/java/org/gradle/internal/logging/progress/DefaultProgressLoggerFactory.java#L55-L55)

Maybe a way forward would be to introduce `BuildOperationIdentifier` now and eventually removed `OperationIdentifier` at a later point if we migrated more things, including the `ProgressLogger` related classes, to build operations.